### DISCUSSION TITLE
Building a Ruleset from Json

### DISCUSSION THREAD

**Discussion by ArunNKutty** [hyperjumptech/grule-rule-engine#280] [Q&A]
package main

import (
	"github.com/hyperjumptech/grule-rule-engine/ast"
	"github.com/hyperjumptech/grule-rule-engine/builder"
	"github.com/hyperjumptech/grule-rule-engine/engine"
	"github.com/hyperjumptech/grule-rule-engine/pkg"
)

func main() {

	dataContext := ast.NewDataContext()

	userData := []byte(`{"name":"John", "age":40,"message":"Hello World"}`)
	err := dataContext.AddJSON("MF", userData)

	if err != nil {
		panic(err)
	}

	knowledgeLibrary := ast.NewKnowledgeLibrary()
	ruleBuilder := builder.NewRuleBuilder(knowledgeLibrary)

	const drls = `{
		"name": "Age Check",
		"desc": "Check for age",
		"salience": 10,
		"when": "MF.Age == 40",
		"then": [
			"Log(\"Age is 40\")"
			]	
	}`

	ruleset, err := pkg.ParseJSONRule([]byte(drls))
	if err != nil {
		panic(err)
	}
	errr := ruleBuilder.BuildRuleFromResource("TutorialRules", "0.0.1", pkg.NewBytesResource([]byte(ruleset)))
	if errr != nil {
    	panic(errr)
	}

	knowledgeBase := knowledgeLibrary.NewKnowledgeBaseInstance("TutorialRules", "0.0.1")

	engine := engine.NewGruleEngine()
	err = engine.Execute(dataContext, knowledgeBase)
	if err != nil {
    	panic(err)
	}
}

I am getting the following error :

ERRO[0000] GRL error, after '[rule Age]' and then unexpected 'Check'  lib=grule struct=GruleParserV3Listener
ERRO[0000] GRL syntax error. got got 1 error(s) in grl the script  lib=grule-rule-engine package=builder
ERRO[0000] 0 : grl error on 1:9 extraneous input 'Check' expecting {'{', SALIENCE, DQUOTA_STRING, SQUOTA_STRING}  lib=grule-rule-engine package=builder

Any help is highly appreciated

**Comment by niallnsec** [comment]
I believe the issue here is that you are using a rule name which has a space in it, which is not allowed in GRL syntax. There should definitely be a more readable error printed though.
